### PR TITLE
Client side autocompletion

### LINKED ISSUE #11: gocode: serverless mode
Currently gocode is a thin client that talks to a server process using net/rpc.

However as far as I can tell the server (unlike the original nsf/gocode version) doesn't cache anything.

A client that uses suggest package to find the list of candidates is going to be simpler and probably even a few ms faster.

I suggest to implement a serverless mode for the client first (e.g. by passing -addr=none on the command line).

If it proves to be fast and usable enough maybe the server can be removed alltogether.

Thoughts?

### PR DESCRIPTION
**Pull request by kostya-sh**
Fixes #11

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -33,30 +48,29 @@ func doClient() {
...
 	client, err := rpc.Dial(*g_sock, addr)
 	if err != nil {
-		if *g_sock == "unix" && fileExists(addr) {
-			os.Remove(addr)
+		if command == "exit" {
+			log.Fatal(err)
 		}
 
+		if *g_sock == "unix" {
+			_ = os.Remove(addr)
```
Comment (mdempsky) [inline client.go]: I assume "_ =" is to satisfy some linter?

File: client.go [hunk 2]
```diff
@@ -12,11 +12,17 @@ import (
...
+	"runtime/debug"
+
 	"github.com/mdempsky/gocode/gbimporter"
 	"github.com/mdempsky/gocode/suggest"
 )
 
 func doClient() {
+	// Client is a short-lived program.
+	// Disable GC to make it faster
+	debug.SetGCPercent(-1)
```
Comment (mdempsky) [inline client.go:24]: Hmm, I'm a little iffy on this, but I guess we can try it out.

**Review by mdempsky — APPROVED** [review]

File: client.go (hunk 1 above)
Comment (kostya-sh) [reply client.go]: The error here is deliberately ignored. As long as tryStartServer() succeeds we do not care if the file deletion failed.

File: client.go (hunk 2 above)
Comment (kostya-sh) [reply client.go:24]: I am not fan of this myself, but it does speed up autocompletion about 30% (from 38-40ms to 26-28ms on a simple test). This actually makes client side autocompletion faster than server side.

File: client.go (hunk 1 above)
Comment (mdempsky) [reply client.go]: Understood.  I'm asking about why it's "_ = os.Remove(addr)" instead of just "os.Remove(addr)".

Comment (kostya-sh) [reply client.go]: yes, this is for https://github.com/kisielk/errcheck

**Comment by LK4D4** [comment]
nice, thanks!

**Pull request merged**